front/app/containers/Admin/reporting/containers/ReportBuilder/index.tsx

```diff
@@ -59,25 +77,28 @@ const ReportBuilder = ({ report, reportLayout }: Props) => {
   const templatePhaseId = search.get('templatePhaseId');
   const previewEnabled = search.get('preview') === 'true';
 
+  const { craftjs_json } = reportLayout.attributes;
+  const isEmptyReport = isEmpty(craftjs_json);
+
+  const [initialData] = useState(() => {
```

Comment:
Is using `useState` without a setter the same as using `useMemo`?

Reply:
It's not the same, because this function inside of `useState` only runs once, while `useMemo` runs every time a dependency changes. In this case, we only need this to run on mount, because afterwards craftjs manages state internally.

Reply:
And if you do not pass any dependencies to `useMemo`, would it be the same? I'm not saying any approach is better, just trying to understand how it works :slightly_smiling_face:

Reply:
Yes, but the linter would start complaining